### PR TITLE
DOC-12075 update repo license

### PR DESCRIPTION
**Pull request by ggray-cb**
Adding a LICENSE file to the 7.0 branch. This will be rolled out to all of the server branches.

Please let me know who welse to include... I assume maybe someone from legal?

### CODE REVIEW

**Comment by osfameron** [comment]
Oh, didn't know we had settled on a CC-BY-NC-SA 4.0 (a great license, no issues with it myself) 👍

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Couchbase Server Documentation © 2024 by Couchbase Inc. is licensed under Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International 
+
+CC BY-NC-SA 4.0.
+
+See https://creativecommons.org/licenses/by-nc-sa/4.0/ for details.
```
Comment (RichardSmedley) [inline LICENSE:3]: Note, in the SDKs we've gone for presenting the complete licence, to save users a further click (away from the site):
https://github.com/couchbase/docs-sdk-scala/blob/release/1.5/LICENSE

And the new IA SDK docs (coming in Q2) have a mention in the README - https://github.com/couchbase/docs-sdk-scala/blob/release/1.6/README.adoc#docs-license

Comment (ggray-cb) [reply LICENSE:3]: I didn't see the SDK docs had the license file already. Searching through github, it does seem common for repos by major companies (i.e. Microsoft) to include the full license text in their LICENSE file, so it makes sense to follow suit.

Comment (osfameron) [reply LICENSE:3]: But creativecommons is pathologically hip and online, and just presenting a URL reflects that

Comment (ggray-cb) [reply LICENSE:3]: Well, if hipness is our goal, we could link to a TikTok video of someone doing an interpretive dance of the CC license text.

Comment (lamagnifica) [reply LICENSE:3]: I second the motion.

Comment (osfameron) [reply LICENSE:3]: This is a proposal we can all get behind. Let's set a date for the next docs summit.

**Review by bfavini — APPROVED** [review]

**Review by lamagnifica — APPROVED** [review]

**Pull request merged**